src/main/scala/uk/gov/hmrc/ct/computations/calculations/AnnualInvestmentAllowanceCalculator.scala

```diff
@@ -24,66 +24,87 @@ import scala.math.BigDecimal.RoundingMode
 
 trait AnnualInvestmentAllowanceCalculator {
 
-  def maximum(cp1: CP1, cp2: CP2, allowableAmounts: Set[AnnualInvestmentAllowancePeriod]) : CATO02 = {
-    val result = allowableAmounts.map { investmentAllowancePeriod =>
-      val daysInPeriod = investmentAllowancePeriod.intersectingDays(cp1.value, cp2.value)
-      val allowanceThisPeriod = BigDecimal(daysInPeriod) * (BigDecimal(investmentAllowancePeriod.maximumAllowed) / yearLengthForApportioning(cp1.value, cp2.value))
+  val DaysInYear   = 365
+  val MonthsInYear = 12
 
-      allowanceThisPeriod.setScale(0, RoundingMode.UP).toInt
-    }.sum
+  def maximum(cp1: CP1, cp2: CP2, allowableAmounts: Set[AnnualInvestmentAllowancePeriod]): CATO02 = {
+    val maximumEntitlement = maximumEntitlementForAccountingPeriod(cp1, cp2, allowableAmounts) _
```

Comment:
I would change this to use two separate overloaded methods instead of passing through a flag and a function. Not as clever but I think it would be more maintainable and easier to read.